lib/Segments/SegmentSaveController.php

```diff
@@ -25,6 +33,33 @@ public function save(array $data = []): SegmentEntity {
     return $this->segmentsRepository->createOrUpdate($name, $description, $id);
   }
 
+  public function duplicate(SegmentEntity $segmentEntity): SegmentEntity {
+    $duplicate = clone $segmentEntity;
+    $duplicate->setName(sprintf(__('Copy of %s', 'mailpoet'), $segmentEntity->getName()));
+
+    $this->checkSegmenUniqueName($duplicate->getName(), $duplicate->getId());
+
+    $this->segmentsRepository->persist($duplicate);
+    $this->segmentsRepository->flush();
+    
+    $subscriberSegments = $this->subscriberSegmentRepository->findBy(['segment' => $segmentEntity]);
+    foreach ($subscriberSegments as $subscriberSegment) {
+      $subscriber = $subscriberSegment->getSubscriber();
+      if (!$subscriber) {
+        continue;
+      }
+      $subscriberDuplicate = new SubscriberSegmentEntity(
+        $duplicate,
+        $subscriber,
+        SubscriberEntity::STATUS_SUBSCRIBED
+      );
+      $this->subscriberSegmentRepository->persist($subscriberDuplicate);
```

Comment:
I am not sure if it is possible or practical, but should we close all of the queries here in a transaction?

Reply:
This is a perfect idea. I thought about it, but I forgot then.